crates/rolldown/src/stages/generate_stage/chunk_optimizer.rs

```diff
@@ -323,4 +336,161 @@ impl GenerateStage<'_> {
       can_merge.then_some(*chunk_idx)
     })
   }
+
+  /// Optimizes facade dynamic entry chunks by merging them with their target common chunks.
+  ///
+  /// This optimization handles the case where a dynamic entry chunk has no modules of its own
+  /// because all its modules were moved to a common chunk(When dynamic entry module are captured by `advancedChunks`). Instead of keeping an empty entry chunk,
+  /// we rewrite references to point directly to the common chunk and ensure proper symbol inclusion.
+  ///
+  /// The function also ensures the runtime module is properly assigned to a chunk when needed.
+  pub(super) fn optimize_facade_dynamic_entry_chunks(
+    &mut self,
+    chunk_graph: &mut ChunkGraph,
+    index_splitting_info: &IndexSplittingInfo,
+    input_base: &ArcStr,
+    module_to_assigned: &mut IndexVec<ModuleIdx, bool>,
+  ) {
+    // Find empty dynamic entry chunks that should be merged with their target common chunks
+    let mut rewrite_entry_to_chunk = FxHashMap::default();
+    let runtime_module_idx = self.link_output.runtime.id();
+    for (chunk_idx, chunk) in chunk_graph.chunk_table.iter_enumerated() {
+      let ChunkKind::EntryPoint { meta, bit: _, module } = chunk.kind else {
+        continue;
+      };
+      if meta.intersects(ChunkMeta::UserDefinedEntry | ChunkMeta::EmittedChunk) {
+        continue;
+      }
+      if !chunk.modules.is_empty() {
+        continue;
+      }
+      // Check if the entry module is included in a common chunk
+      let Some(target_chunk_idx) = chunk_graph.module_to_chunk[module] else {
+        continue;
+      };
+      let target_chunk = &chunk_graph.chunk_table[target_chunk_idx];
+      if !matches!(target_chunk.kind, ChunkKind::Common) {
+        continue;
+      }
+      if !matches!(target_chunk.chunk_reason_type.as_ref(), ChunkReasonType::AdvancedChunks { .. })
+      {
+        continue;
+      }
+      rewrite_entry_to_chunk.insert(module, (chunk_idx, target_chunk_idx));
+    }
+
+    // Namespace symbols by default reference all exported symbols from the module.
+    // To preserve dynamic import tree shaking, we should only include symbols that were actually used during the linking stage.
+    // This ensures that including a namespace symbol doesn't inadvertently add new exported symbols.
```

Comment:
```suggestion
    // This ensures that including a namespace symbol doesn't inadvertently add unused exported symbols.
```
I don't understand what "new" means here. Is it the unused symbols?